src/spikeinterface/core/generate.py

```diff
@@ -220,16 +221,340 @@ def add_synchrony_to_sorting(sorting, sync_event_ratio=0.3, seed=None):
+            self._cached_spike_vector = np.concatenate((self._cached_spike_vector, added_spikes_existing_units))
+            self.added_spikes_from_existing_mask = np.concatenate(
+                (self.added_spikes_from_existing_mask, np.ones(len(added_spikes_existing_units), dtype=bool))
+            )
+            self.added_spikes_from_new_mask = np.concatenate(
+                (self.added_spikes_from_new_mask, np.zeros(len(added_spikes_existing_units), dtype=bool))
+            )
+
+        if added_spikes_new_units is not None and len(added_spikes_new_units) > 0:
+            assert (
+                added_spikes_new_units.dtype == minimum_spike_dtype
+            ), "added_spikes_new_units should be a spike vector"
+            self._cached_spike_vector = np.concatenate((self._cached_spike_vector, added_spikes_new_units))
+            self.added_spikes_from_existing_mask = np.concatenate(
+                (self.added_spikes_from_existing_mask, np.ones(len(added_spikes_new_units), dtype=bool))
+            )
+            self.added_spikes_from_new_mask = np.concatenate(
+                (self.added_spikes_from_new_mask, np.ones(len(added_spikes_new_units), dtype=bool))
+            )
+
+        sort_idxs = np.lexsort([self._cached_spike_vector["sample_index"], self._cached_spike_vector["segment_index"]])
+        self._cached_spike_vector = self._cached_spike_vector[sort_idxs]
+        self.added_spikes_from_existing_mask = self.added_spikes_from_existing_mask[sort_idxs]
+        self.added_spikes_from_new_mask = self.added_spikes_from_new_mask[sort_idxs]
+
+        # We need to add the sorting segments
+        for segment_index in range(sorting.get_num_segments()):
+            segment = NumpySortingSegment(self._cached_spike_vector, segment_index, unit_ids=self.unit_ids)
+            self.add_sorting_segment(segment)
+
+        if self.refractory_period_ms is not None:
+            self.clean_refractory_period()
+
+        self._kwargs = dict(
+            sorting=sorting,
+            added_spikes_existing_units=added_spikes_existing_units,
+            added_spikes_new_units=added_spikes_new_units,
+            new_unit_ids=new_unit_ids,
+            refractory_period_ms=refractory_period_ms,
+        )
+
+    @property
+    def added_spikes_mask(self):
+        return np.logical_or(self.added_spikes_from_existing_mask, self.added_spikes_from_new_mask)
+
+    def get_added_spikes_indices(self):
+        return np.nonzero(self.added_spikes_mask)[0]
+
+    def get_added_spikes_from_existing_indices(self):
+        return np.nonzero(self.added_spikes_from_existing_mask)[0]
+
+    def get_added_spikes_from_new_indices(self):
+        return np.nonzero(self.added_spikes_from_new_mask)[0]
+
+    def get_added_units_inds(self):
+        indices = self._cached_spike_vector["unit_index"][self.get_added_spikes_from_new_indices()]
+        return np.unique(self.unit_ids[indices])
```

Comment:
Sam remark as before.
We could have it as an attr

Reply:
I can make an attr, we I thought you wanted methods (and there are methods everywhere). Plus here this is not the same as before. If you have added spikes from new units, but with units with no spikes, then the fact that these inds are not returned by get_added_units_inds is not shocking to me. You have no spikes for them... How to you want to return them. You can do what you are suggesting by returning self.unit_ids[len(self.parent_sorting.unit_ids):], but it would requires saving parent_sorting in the __init__ as an attribute. This was currently not the case

Reply:
I've added the attribute self.parent_unit_ids as suggested